Scripts/CodeFormat/check_format.sh

```diff
@@ -1,9 +1,20 @@
 #!/usr/bin/env bash
 
-SOURCE_COMMIT="$1"
-if [ "$#" -gt 0 ]; then
-    shift
-fi
+# Parse arguments
+ALL_FILES=false
+SOURCE_COMMIT=""
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --all-files)
+            ALL_FILES=true
+            shift
+            ;;
+        *)
+            SOURCE_COMMIT="$1"
+            shift
+            ;;
+    esac
+done
```

Comment:
Just a last comment here :). If this is a while loop then in the case when we don't specify the `--all-files` all the arguments are shifted (because we always go to the default case) and then we end up with no arguments to pass to `git-clang-format` and with `SOURCE_COMMIT` being the last argument specified, which is not the expected behaviour. As I see it, right now the logic should be:
- if all files should be formatted/checked, specify `--all-files` as the first argument
- if only the modified files should be formatted/checked, specify a base revision commit as the first argument (which was the previous behaviour)

So I think this while should be instead something along the lines of:
```suggestion
# Parse arguments
ALL_FILES=false
SOURCE_COMMIT=""
# Either a base SHA is used or all files are checked
if [[ $# -gt 0 ]]; then
    case "$1" in
        "--all-files")
            ALL_FILES=true
            ;;
        *)
            SOURCE_COMMIT=$1
            ;;
    esac
    shift
fi
```
what do you think?

Reply:
Awesome, thanks!